File: app/src/main/java/com/sjn/stamp/db/RankedSong.java
package com.sjn.stamp.db;

import android.content.res.Resources;

import com.sjn.stamp.R;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(prefix = "m")
@AllArgsConstructor(suppressConstructorProperties = true)
@NoArgsConstructor
@Getter
@Setter
public class RankedSong implements Shareable {
    int mPlayCount;
    public Song mSong;

    @Override
    public String share(Resources resources) {
        if (mSong == null) {
            return "";
        }
        return resources.getString(R.string.share_ranked, mPlayCount, mSong.share(resources));
    }
}

File: app/src/main/java/com/sjn/stamp/db/RankedSong.kt
package com.sjn.stamp.db

import android.content.res.Resources
import com.sjn.stamp.R

class RankedSong(
        var playCount: Int,
        var song: Song
) : Shareable {
    override fun share(resources: Resources): String = resources.getString(R.string.share_ranked, playCount, song.share(resources))
}
